core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -1101,6 +1112,54 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
     return n;
   }
 
+  template <class TagType>
+  struct ThrustFunctorWrapper {
+    const FunctorType& f;
+
+    KOKKOS_FUNCTION
+    ThrustFunctorWrapper(const FunctorType& op) : f(op){};
+
+    template <class TagType_                      = TagType,
+              typename std::enable_if<std::is_same<TagType_, void>::value,
+                                      bool>::type = true>
+    KOKKOS_FUNCTION value_type operator()(index_type i) const {
+      // value_type val = InitValueType();
+      value_type val = (value_type)0;  // for now, assuming no init value
+      f(i, val);
+      return val;
+    }
+
+    template <class TagType_                      = TagType,
+              typename std::enable_if<!std::is_same<TagType_, void>::value,
+                                      bool>::type = true>
+    KOKKOS_FUNCTION value_type operator()(index_type i) const {
+      // value_type val = InitValueType();
+      value_type val = (value_type)0;  // for now, assuming no init value
+      f(TagType(), i, val);
+      return val;
+    }
+  };
+
+  inline void thrust_execute() {
+    printf("using CUDA Thrust\n");
+
+    thrust::counting_iterator<value_type> temp_iter_d(m_policy.begin());
```

Comment:
no this needs to be just a counting iterator of int or int64_t or even better the policies index type. Not the value type.

Reply:
Thanks for the catch, implemented with index_type, pushing commit